docs(dialog):documentation for the dialog component (#617)
telerik/blazor-docs: components/dialog/overview.md

---
title: Overview
page_title: Dialog Overview
description: Overview of the Dialog for Blazor.
slug: dialog-overview
tags: telerik,blazor,dialog,overview
published: True
position: 0
---

# Dialog Overview

> The Dialog is a modal popup that brings information to the user.

Reviewer: COnsider combining the first two paragraphs into one, as it is the first one is quite short and give rather little info

---
title: Overview
page_title: Dialog Overview
description: Overview of the Dialog for Blazor.
slug: dialog-overview
tags: telerik,blazor,dialog,overview
published: True
position: 0
---

# Dialog Overview

> The Dialog component is a modal popup that displays information to the user. It is fully customizable and extends the `DialogBuilder` to provide you with a way to achieve complex UI popups.

Reviewer 2: * This introduction should highlight the core purpose and most valuable feature of the component. It should answer the question "Why / when should I use it?" What is the difference between the Dialog and the Window?
* "fully customizable" - marketing phrase, doesn't mean anything
* "Displays infomation to the user" is too generic - each of our components does that.
* "Extends the DialogBuilder" - this is an implementation detail that may not belong here, unless it's really important.
* "complex UI popups" - what is a complex popup?

## Basics

To add a Telerik Dialog to your Blazor app:

1. Add the `TelerikDialog` tag.
1. Provide `Visible` parameter.

> 1. Set its `Title`.

Reviewer 2: ```suggestion
1. Set a `Title`.
```

1. Optionally, choose a `ButtonsLayout` (from the `DialogButtonsLayout` enums).

>     * Its default layout is `DialogButtonsLayout.Stretched`. See more in the [Action Buttons article]({%slug  dialog-action-buttons%})).

Reviewer 2: These steps should be simple. Move all explanations to the Features section and make "choose a ButtonLayout" a link to it. Is it ButtonsLayout or ButtonLayout - both syntaxes are used.

Author: Done.

1. Optionally, set its `ShowCloseButton` parameter (true/false).

>     * Its **true** by default. See more in the [Header article]({%slug  dialog-header%}).

Reviewer 2: Consider skipping this step for simplicity.

> The following example demonstrates how to set up the Dialog with its default configuration.

Reviewer 2: Make the example code match the suggestions in the steps.

>caption A basic configuration of the Telerik Dialog.

````CSHTML
@* An example of the Dialog basic implementation. *@

<TelerikDialog @bind-Visible="@Visible"
               Title="@Title">
    <DialogContent>
        <br />

>         <div>--- Place here the Dialog content ---</div>

Reviewer 2: Think of a simple real scenario, instead of dummy text.

        <br />
    </DialogContent>
    <DialogButtons>
        <TelerikButton Primary="true" OnClick="@(() => { Visible = false; })">Cancel</TelerikButton>

>         <TelerikButton OnClick="@(() => { Visible = false; })">OK</TelerikButton>

Reviewer 2: A primary "cancel" button is suitable only for destructive operations and may look strange here. Consider using a standard example with primary "OK".

    </DialogButtons>
</TelerikDialog>

@code {
    private bool Visible { get; set; } = true;
    private string Title { get; set; } = "Title here";
}
````

>caption The result from the above code snippet.

![](images/dialog-basic-overview.png)

> ## Features

Reviewer 2: Sort and consider grouping the attributes by importance and topic.

> >caption The Dialog provides the following features:

Reviewer 2: This can be a standard sentence, instead of a caption.

* `Title` - `string` - defines the title of the Dialog.

* `Visible` - `bool` - defines the visibility of the Dialog.

* `DialogTitle` - `RenderFragment` - defines the title template of the component.

* `DialogContent` - `RenderFragment` - defines the content template of the component.

* `DialogButtons` - `RenderFragment` - defines the actions bar template of the component.

* `ButtonLayout` - `enum`- `DialogButtonsLayout` - defines the layout of the actions button in the footer.

> * `VisibleChanged` - `EventCallback<false>` - triggers when the visibility of the component changes.

Reviewer 2: Shouldn't this be in a separate Events article?

* `Class` - `string` - defines the class of the component instance.

* `Width` - `string` - defines the width of the Dialog.

* `Height` - `string` - defines the height of the Dialog.

* `ShowCloseButton` - `bool` - defines the close behavior of the component - whether the component should render close flat button in the titlebar.

> * `CloseOnOverlayClick` - `bool` - defines whether clicking the overlay should close the dialog.

Reviewer 2: ```suggestion
* `CloseOnOverlayClick` - `bool` - defines whether clicking on the modal overlay should close the Dialog.
```

> * `FocusedElementSelector` - `string` - defines the queryselector of the initially focused item on open. By default, it is the first focusable item in the dialog.

Reviewer 2: queryselector is a JavaScript term and may not be known to Blazor devs. I assume "CSS selector" is the same, right?

```suggestion
* `FocusedElementSelector` - `string` - defines the CSS selector of the initially focused item on open. By default, it is the first focusable item in the dialog.
```
